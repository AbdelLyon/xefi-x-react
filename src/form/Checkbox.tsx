import { forwardRef } from "react";
import type { CheckboxGroupProps, CheckboxProps } from "@heroui/react";
import { CheckboxGroup as CheckboxGroupRoot, Checkbox } from "@heroui/react";
import { mergeTailwindClasses } from "@/utils/utils";

type CheckboxItemProps = {
  label?: React.ReactNode;
} & Omit<CheckboxProps, "children">;

type CheckboxWrapperProps = {
  items: CheckboxItemProps[];
  groupClasses?: {
    base?: string;
    label?: string;
  };
  itemClasses?: {
    base?: string;
    label?: string;
    wrapper?: string;
  };
} & Omit<CheckboxGroupProps, "children">;

const CheckboxGroup = forwardRef<HTMLDivElement, CheckboxWrapperProps>(
  (
    {
      items,
      groupClasses,
      itemClasses,
      label = "Select options",
      defaultValue,
      ...props
    },
    ref,
  ): React.ReactNode => {
    const defaultGroupClasses = {
      base: "w-full",
      label: "text-medium font-semibold",
    };

    const defaultItemClasses = {
      base: "w-full",
      label: "text-small",
      wrapper: "",
    };

    return (
      <CheckboxGroupRoot
        ref={ref}
        label={label}
        defaultValue={defaultValue}
        {...props}
        classNames={{
          base: mergeTailwindClasses(
            defaultGroupClasses.base,
            groupClasses?.base,
          ),
          label: mergeTailwindClasses(
            defaultGroupClasses.label,
            groupClasses?.label,
          ),
        }}
      >
        {items.map(
          (item): React.ReactNode => (
            <Checkbox
              key={item.value}
              {...item}
              classNames={{
                base: mergeTailwindClasses(
                  defaultItemClasses.base,
                  itemClasses?.base,
                  item.className,
                ),
                label: mergeTailwindClasses(
                  defaultItemClasses.label,
                  itemClasses?.label,
                  item.classNames?.label,
                ),
                wrapper: mergeTailwindClasses(
                  defaultItemClasses.wrapper,
                  itemClasses?.wrapper,
                  item.classNames?.wrapper,
                ),
              }}
            >
              {item.label}
            </Checkbox>
          ),
        )}
      </CheckboxGroupRoot>
    );
  },
);

CheckboxGroup.displayName = "CheckboxGroup";

export { Checkbox, CheckboxGroup };
