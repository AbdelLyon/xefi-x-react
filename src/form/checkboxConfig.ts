/**
 * Checkbox component configuration
 * Provides standardized styling and behavior patterns
 */

import { createClassNamesConfig } from "@/utils/classNames";
import { baseFormFieldClasses, type FormFieldConfig, type FormGroupConfig } from "./formConfig";
import type { ReactNode } from "react";

/**
 * Default class names for checkbox components
 */
export const defaultCheckboxClassNames = {
  ...baseFormFieldClasses,
  base: "group relative max-w-fit inline-flex items-center justify-start cursor-pointer",
  wrapper: "relative inline-flex items-center justify-center flex-shrink-0",
  icon: "z-10 w-4 h-4 opacity-0 group-data-[selected=true]:opacity-100 transition-opacity",
  label: "ml-2 text-foreground select-none",
} as const;

/**
 * Checkbox size class configurations
 */
export const checkboxSizeClasses = {
  sm: {
    wrapper: "w-4 h-4 mr-2",
    icon: "w-3 h-3",
    label: "text-sm",
  },
  md: {
    wrapper: "w-5 h-5 mr-2",
    icon: "w-3 h-3",
    label: "text-base",
  },
  lg: {
    wrapper: "w-6 h-6 mr-3",
    icon: "w-4 h-4",
    label: "text-lg",
  },
} as const;

/**
 * Checkbox color class configurations
 */
export const checkboxColorClasses = {
  default: {
    wrapper: "border-default-300 data-[selected=true]:border-default data-[selected=true]:bg-default",
    icon: "text-default-foreground",
  },
  primary: {
    wrapper: "border-default-300 data-[selected=true]:border-primary data-[selected=true]:bg-primary",
    icon: "text-primary-foreground",
  },
  secondary: {
    wrapper: "border-default-300 data-[selected=true]:border-secondary data-[selected=true]:bg-secondary",
    icon: "text-secondary-foreground",
  },
  success: {
    wrapper: "border-default-300 data-[selected=true]:border-success data-[selected=true]:bg-success",
    icon: "text-success-foreground",
  },
  warning: {
    wrapper: "border-default-300 data-[selected=true]:border-warning data-[selected=true]:bg-warning",
    icon: "text-warning-foreground",
  },
  danger: {
    wrapper: "border-default-300 data-[selected=true]:border-danger data-[selected=true]:bg-danger",
    icon: "text-danger-foreground",
  },
} as const;

/**
 * Checkbox state classes
 */
export const checkboxStateClasses = {
  default: {
    base: "",
    wrapper: "",
  },
  invalid: {
    base: "",
    wrapper: "border-danger data-[selected=true]:border-danger data-[selected=true]:bg-danger",
  },
  valid: {
    base: "",
    wrapper: "border-success data-[selected=true]:border-success data-[selected=true]:bg-success",
  },
  disabled: {
    base: "opacity-50 cursor-not-allowed",
    wrapper: "border-default-200 bg-default-100",
  },
} as const;

/**
 * Create checkbox class configuration
 */
export const checkboxClassConfig = createClassNamesConfig(defaultCheckboxClassNames);

/**
 * Get checkbox size classes
 */
export const getCheckboxSizeClasses = (size: keyof typeof checkboxSizeClasses = "md") => {
  return checkboxSizeClasses[size] || checkboxSizeClasses.md;
};

/**
 * Get checkbox color classes
 */
export const getCheckboxColorClasses = (color: keyof typeof checkboxColorClasses = "primary") => {
  return checkboxColorClasses[color] || checkboxColorClasses.primary;
};

/**
 * Get checkbox state classes
 */
export const getCheckboxStateClasses = (state: keyof typeof checkboxStateClasses = "default") => {
  return checkboxStateClasses[state] || checkboxStateClasses.default;
};

/**
 * Checkbox item configuration
 */
export interface CheckboxItemConfig extends FormFieldConfig {
  /** Checkbox value */
  value: string;
  /** Checkbox label content */
  label?: ReactNode;
  /** Whether checkbox is checked */
  checked?: boolean;
  /** Whether checkbox is indeterminate */
  indeterminate?: boolean;
  /** Custom icon for checked state */
  icon?: ReactNode;
  /** Custom color theme */
  color?: keyof typeof checkboxColorClasses;
  /** Change handler */
  onChange?: (checked: boolean) => void;
}

/**
 * Checkbox group configuration
 */
export interface CheckboxGroupConfig extends FormGroupConfig {
  /** Array of checkbox items */
  items: CheckboxItemConfig[];
  /** Current selected values */
  value?: string[];
  /** Default selected values */
  defaultValue?: string[];
  /** Selection change handler */
  onChange?: (values: string[]) => void;
  /** Maximum selections allowed */
  maxSelections?: number;
  /** Minimum selections required */
  minSelections?: number;
  /** Whether to show select all option */
  showSelectAll?: boolean;
  /** Select all label */
  selectAllLabel?: ReactNode;
}

/**
 * Validate checkbox group selection
 */
export const validateCheckboxGroup = (
  selectedValues: string[],
  config: CheckboxGroupConfig
): { valid: boolean; errors: string[]; } => {
  const errors: string[] = [];

  // Check minimum selections
  if (config.minSelections && selectedValues.length < config.minSelections) {
    errors.push(`Please select at least ${config.minSelections} option${config.minSelections > 1 ? "s" : ""}`);
  }

  // Check maximum selections
  if (config.maxSelections && selectedValues.length > config.maxSelections) {
    errors.push(`Please select no more than ${config.maxSelections} option${config.maxSelections > 1 ? "s" : ""}`);
  }

  // Check required validation
  if (config.required && selectedValues.length === 0) {
    errors.push("Please select at least one option");
  }

  return {
    valid: errors.length === 0,
    errors,
  };
};

/**
 * Create checkbox props with defaults
 */
export const createCheckboxProps = <T extends Record<string, unknown>>(
  userProps: T,
  config: CheckboxItemConfig
) => {
  const sizeClasses = getCheckboxSizeClasses(config.size);
  const colorClasses = getCheckboxColorClasses(config.color);
  const stateClasses = getCheckboxStateClasses(config.state === 'loading' ? 'default' : config.state);

  const mergedClasses = checkboxClassConfig.mergeClasses({
    ...sizeClasses,
    ...colorClasses,
    ...stateClasses,
    ...config.classNames,
  });

  return {
    value: config.value,
    isSelected: config.checked || false,
    isIndeterminate: config.indeterminate || false,
    isRequired: config.required || false,
    isDisabled: config.disabled || false,
    isInvalid: config.state === "invalid",
    size: config.size || "md",
    color: config.color || "primary",
    classNames: mergedClasses,
    children: config.label,
    icon: config.icon,
    onValueChange: config.onChange,
    ...userProps,
  };
};

/**
 * Default checkbox behavior
 */
export const defaultCheckboxBehavior = {
  size: "md" as const,
  color: "primary" as const,
  state: "default" as const,
  required: false,
  disabled: false,
  checked: false,
  indeterminate: false,
} as const;