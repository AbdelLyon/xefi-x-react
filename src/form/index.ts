import "@/index.css";
export { RadioGroup } from "@/form/Radio";
export { Checkbox, CheckboxGroup } from "@/form/Checkbox";
export { InputOtp } from "@/form/InputOpt";
export { Input } from "@/form/Input";
export { Textarea } from "@/form/Textarea";
export { Switch } from "@/form/Switch";
export { Select, type SelectOption } from "@/form/Select";
export { InfiniteSelect } from "@/form/InfiniteSelect";
export { InfiniteAutocomplete } from "@/form/InfiniteAutocomplete";
