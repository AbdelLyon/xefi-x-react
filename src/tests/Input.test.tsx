import { Input } from "@/form";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

describe("Composant Input", (): void => {
  describe("Snapshots", (): void => {
    it("devrait correspondre au snapshot avec les props par défaut", (): void => {
      const { container } = render(<Input />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot pour chaque variante", (): void => {
      const variants = ["bordered", "flat", "faded", "underlined"] as const;
      variants.forEach((variant): void => {
        const { container } = render(<Input variant={variant} />);
        expect(container).toMatchSnapshot();
      });
    });

    it("devrait correspondre au snapshot pour différentes tailles", (): void => {
      const sizes = ["sm", "md", "lg"] as const;
      sizes.forEach((size): void => {
        const { container } = render(<Input size={size} />);
        expect(container).toMatchSnapshot();
      });
    });

    it("devrait correspondre au snapshot pour le type password", (): void => {
      const { container } = render(<Input type="password" />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec label", (): void => {
      const { container } = render(<Input label="Nom d'utilisateur" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("Comportement", (): void => {
    it("devrait basculer la visibilité du mot de passe", (): void => {
      const { container } = render(<Input type="password" />);
      const input = container.querySelector(
        'input[type="password"]',
      ) as HTMLInputElement;
      const toggleButton = screen.getByRole("button");

      expect(input.type).toBe("password");
      fireEvent.click(toggleButton);
      expect(input.type).toBe("text");
      fireEvent.click(toggleButton);
      expect(input.type).toBe("password");
    });

    it("devrait appeler la fonction customValidation", (): void => {
      const customValidation = vi.fn().mockReturnValue(true);
      const { container } = render(
        <Input customValidation={customValidation} />,
      );
      const input = container.querySelector("input") as HTMLInputElement;
      fireEvent.change(input, { target: { value: "test" } });
      expect(customValidation).toHaveBeenCalledWith("test");
    });

    it("devrait être désactivé quand isDisabled est true", (): void => {
      const { container } = render(<Input isDisabled />);
      const input = container.querySelector("input") as HTMLInputElement;
      expect(input.disabled).toBe(true);
    });
  });

  describe("Accessibilité", (): void => {
    it("devrait avoir l'attribut required quand isRequired est true", (): void => {
      const { container } = render(<Input isRequired />);
      const input = container.querySelector("input") as HTMLInputElement;
      expect(input.required).toBe(true);
    });

    it("devrait avoir l'attribut readonly quand isReadOnly est true", (): void => {
      const { container } = render(<Input isReadOnly />);
      const input = container.querySelector("input") as HTMLInputElement;
      expect(input.readOnly).toBe(true);
    });
  });
});
