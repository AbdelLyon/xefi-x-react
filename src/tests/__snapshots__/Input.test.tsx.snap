// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composant Input > Snapshots > devrait correspondre au snapshot avec label 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full"
    data-has-elements="true"
    data-has-label="true"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="relative w-full inline-flex tap-highlight-transparent shadow-sm px-3 min-h-10 rounded-medium flex-col items-start justify-center gap-0 transition-background !duration-150 transition-colors motion-reduce:transition-none py-2 border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
      data-slot="input-wrapper"
      style="cursor: text;"
    >
      <label
        class="absolute z-10 pointer-events-none origin-top-left flex-shrink-0 rtl:origin-top-right subpixel-antialiased block text-foreground-500 cursor-text will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-default-600 group-data-[filled-within=true]:pointer-events-auto group-data-[filled-within=true]:scale-85 text-small group-data-[filled-within=true]:-translate-y-[calc(50%_+_theme(fontSize.small)/2_-_6px_-_theme(borderWidth.medium))] pe-2 max-w-full text-ellipsis overflow-hidden"
        data-slot="label"
        for="react-aria-:r1d:"
        id="react-aria-:r1e:"
      >
        Nom d'utilisateur
      </label>
      <div
        class="inline-flex w-full items-center h-full box-border group-data-[has-label=true]:items-end"
        data-slot="inner-wrapper"
      >
        <input
          aria-label="Nom d'utilisateur"
          aria-labelledby="react-aria-:r1d: react-aria-:r1e:"
          class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
          data-slot="input"
          id="react-aria-:r1d:"
          tabindex="0"
          title=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot avec les props par d√©faut 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-10 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
            data-slot="input"
            id="react-aria-:r0:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour chaque variante 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-10 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
            data-slot="input"
            id="react-aria-:r5:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour chaque variante 2`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background border-none bg-default-100 dark:bg-default-50 data-[hover=true]:bg-content1-300-200 group-data-[focus=true]:bg-default-100 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small group-data-[has-value=true]:text-default-foreground"
            data-slot="input"
            id="react-aria-:ra:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour chaque variante 3`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 data-[hover=true]:border-default-400 focus-within:border-default-400 min-h-10 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background border-1 border-transparent bg-default-100 dark:bg-default-50 data-[hover=true]:bg-content1-300-200 group-data-[focus=true]:border-outline h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
            data-slot="input"
            id="react-aria-:rf:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour chaque variante 4`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="w-full inline-flex tap-highlight-transparent flex-row items-center px-3 gap-3 !px-1 !pb-0 !gap-0 box-border shadow-[0_1px_0px_0_rgba(0,0,0,0.05)] !rounded-none hover:border-default-300 after:content-[''] after:w-0 after:origin-center after:absolute after:left-1/2 after:-translate-x-1/2 after:-bottom-[2px] after:h-[2px] group-data-[focus=true]:after:w-full min-h-10 transition-background motion-reduce:transition-none !duration-150 after:transition-width motion-reduce:after:transition-none relative border-b-1 rounded-none bg-transparent border-border h-12 after:bg-outline data-[hover=true]:after:scale-x-100 data-[hover=true]:after:bg-outline group-data-[focus=true]:after:scale-x-100 group-data-[focus=true]:after:bg-outline"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border pb-1.5"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small group-data-[has-value=true]:text-foreground"
            data-slot="input"
            id="react-aria-:rk:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour diff√©rentes tailles 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_8px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-8 px-2 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
            data-slot="input"
            id="react-aria-:rp:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour diff√©rentes tailles 2`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-10 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small"
            data-slot="input"
            id="react-aria-:ru:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour diff√©rentes tailles 3`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-12 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium"
            data-slot="input"
            id="react-aria-:r13:"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant Input > Snapshots > devrait correspondre au snapshot pour le type password 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="h-full flex flex-col"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 min-h-10 rounded-medium transition-background !duration-150 transition-colors motion-reduce:transition-none border-1 bg-white dark:bg-background border-border data-[hover=true]:border-outline group-data-[focus=true]:border-outline group-data-[focus=true]:bg-content1 h-12"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small [&::-ms-reveal]:hidden"
            data-has-end-content="true"
            data-slot="input"
            data-type="password"
            id="react-aria-:r18:"
            tabindex="0"
            title=""
            type="password"
            value=""
          />
          <button
            class="opacity-40 focus:outline-none"
            type="button"
          >
            <svg
              class="tabler-icon tabler-icon-eye pointer-events-none"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0"
              />
              <path
                d="M21 12c-2.4 4 -5.4 6 -9 6c-3.6 0 -6.6 -2 -9 -6c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
